buzzard, rxjs 5.5

j2L4e opened the pull request:
Made everything ready for buzzard,
moved to rxjs 5.5 and pipeable operators

closes #83, #76, #77

daffl commented on src/resource.js:
```diff
@@ -22,29 +34,35 @@ module.exports = function (settings, method) {
+        const filterFn = current => current[options.idField] === data[options.idField];
         // `removed` events get special treatment
-        const filteredRemoves = this.removed$.filter(filter);
+        const filteredRemoves$ = this.removed$.pipe(filter(filterFn));
```
No deal breaker, personally I'm not a big fan of hungarian notation style flags (here the `$`) in variable names.

j2L4e replied:
Yeah, sorry, I shouldn't mess with your naming conventions.
Observables are the only place where I use to do it, anyway. We can revert those.

j2L4e commented on test/resource.test.js:
```diff
@@ -109,32 +126,53 @@ describe('reactive resources', () => {
       }, done);
 
-      result.skip(1).first().subscribe(message => {
+      result.pipe(
```
The old one is still there until rxjs 6.

The new one is pretty nice. You can pass in any function of type Observable => Observable, so composing your own operators is a piece of cake. Also there was the dependency problem with the prototype based operators. 
You can read up on it here https://github.com/ReactiveX/rxjs/blob/master/doc/pipeable-operators.md

j2L4e commented on src/resource.js:
```diff
@@ -22,29 +34,35 @@ module.exports = function (settings, method) {
 
     // apply `let` function if set
-    const letStream = options.let ? stream.let(options.let) : stream;
+    const letStream = options.let ? stream.pipe(options.let) : stream;
```
I think `options.let` should become `options.pipe` for consistency. Also it would encourage using the new style operators and this is major version anyway

daffl replied:
Done, I updated the PR.

j2L4e commented:
There are quite a bunch of hungarians left, want me to get rid of those?

```
$ egrep -Ron '[a-zA-Z]+\$' src
src/index.js:56:created$
src/index.js:57:updated$
src/index.js:58:patched$
src/index.js:59:removed$
src/resource.js:42:removed$
src/resource.js:45:created$
src/resource.js:46:updated$
src/resource.js:47:patched$
src/strategies.js:14:source$
src/strategies.js:15:source$
src/strategies.js:18:source$
src/strategies.js:25:events$
src/strategies.js:26:created$
src/strategies.js:29:removed$
src/strategies.js:30:updated$
src/strategies.js:31:patched$
src/strategies.js:34:source$
src/strategies.js:36:events$
src/strategies.js:37:source$
src/strategies.js:42:source$
src/strategies.js:104:events$
src/strategies.js:105:created$
src/strategies.js:109:removed$
src/strategies.js:112:updated$
src/strategies.js:112:patched$
src/strategies.js:117:source$
src/strategies.js:121:events$
```

daffl commented:
We can leave those for now since they are attached to the service object.
